src/resolvers/taxonomyResolvers.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { IArticleV2 } from "@ndla/types-backend/article-api";
+import { ISubjectPageData } from "@ndla/types-backend/frontpage-api";
+import { Node, TaxonomyContext } from "@ndla/types-taxonomy";
+import {
+  fetchArticle,
+  fetchChildren,
+  fetchLK20CompetenceGoalSet,
+  fetchLearningpath,
+  fetchNode,
+  fetchNodeByContentUri,
+  fetchNodeResources,
+  queryNodes,
+} from "../api";
+import {
+  GQLLearningpath,
+  GQLNode,
+  GQLNodeChildrenArgs,
+  GQLQueryNodeArgs,
+  GQLQueryNodeByArticleIdArgs,
+  GQLQueryNodesArgs,
+  GQLTaxonomyContext,
+  GQLTaxonomyCrumb,
+  GQLTaxonomyEntity,
+  GQLWithArticle,
+} from "../types/schema";
+import { nodeToTaxonomyEntity } from "../utils/apiHelpers";
+import { filterMissingArticles, getArticleIdFromUrn, getLearningpathIdFromUrn } from "../utils/articleHelpers";
+
+const pickContextValues = (node: Node, language: string, rootId?: string, parentId?: string) => {
+  const visibleContexts = node.contexts.filter((c) => c.isVisible && !c.rootId.startsWith("urn:programme"));
+  // A resource can have several context in one subject. rootId and parentId helps picking the right one.
+  const rootContexts = rootId ? visibleContexts.filter((c) => c.rootId === rootId) : visibleContexts;
+  const parentContexts = parentId ? rootContexts.filter((c) => c.parentIds.includes(parentId)) : rootContexts;
+
+  const selectedCtx = parentContexts?.[0];
+  const path = selectedCtx?.path || node.path;
+  const rank = selectedCtx?.rank;
+  const contextId = selectedCtx?.contextId;
+  const relevanceId = selectedCtx?.relevanceId;
+  const entity = nodeToTaxonomyEntity({ ...node, contexts: visibleContexts }, language);
+  return { ...entity, contextId, path, rank, relevanceId };
+};
+
+export const Query = {
+  async node(
+    _: any,
+    { id, rootId, parentId, contextId }: GQLQueryNodeArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode | undefined> {
+    if (id) {
+      const node = await context.loaders.nodeLoader.load({ id, rootId, parentId });
+      return pickContextValues(node, context.language, rootId, parentId);
+    }
+    if (contextId) {
+      const nodes = await queryNodes({ contextId, includeContexts: true, filterProgrammes: true }, context);
+      if (nodes.length === 0) {
+        throw new Error(`No node found with contextId: ${contextId}`);
+      }
+      const entities = nodes.map((node) => nodeToTaxonomyEntity(node, context.language, contextId));
```

Comment:
Pirk, men vi trenger vel strengt tatt bare å transformere den første noden.

Reply:
Det er sant, men i nesten 100% av tilfeller er arrayet på 1. Det er kun en teoretisk mulighet for at du får meir enn 1 treff.

Reply:
Forenkler litt.